### PR TITLE
Tests for BBCode

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp to #6835 

@MrPetovan the test for Bug #6095 fails, so I skipped this test for now, but we should fix the issue behdind it..

And I'd like to remove some of the static dependencies in the develop later, so I can remove the separate process annotations (which is a performance issue for testing)

### CODE REVIEW

File: tests/src/Content/Text/BBCodeTest.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+				'data' => 'https://de.wikipedia.org/wiki/Juha_Sipilä',
+				'link' => true,
+			],
+			'bug-2487-2' => [
+				'data' => 'https://de.wikipedia.org/wiki/Dnepr_(Motorradmarke)',
+				'link' => true,
+			],
+			'bug-2487-3' => [
+				'data' => 'https://friendica.wäckerlin.ch/friendica',
+				'link' => false,
```
Comment (MrPetovan) [inline tests/src/Content/Text/BBCodeTest.php]: This should be true, this is a valid link.

File: tests/src/Content/Text/BBCodeTest.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+			],
+			'bug-2487-4' => [
+				'data' => 'https://mastodon.social/@morevnaproject',
+				'link' => false,
```
Comment (MrPetovan) [inline tests/src/Content/Text/BBCodeTest.php]: This should be true, this is a valid link.

**Comment by MrPetovan** [comment]
Can I run these test without a database set up?

File: tests/src/Content/Text/BBCodeTest.php (hunk 1 above)
Comment (nupplaphil) [reply tests/src/Content/Text/BBCodeTest.php]: when I set it to true:
```console
Failed asserting that two strings are equal.
Expected :'<a href="https://friendica.wäckerlin.ch/friendica" target="_blank">https://friendica.wäckerlin.ch/friendica</a>'
Actual   :'https://friendica.wäckerlin.ch/friendica'
```

File: tests/src/Content/Text/BBCodeTest.php (hunk 2 above)
Comment (nupplaphil) [reply tests/src/Content/Text/BBCodeTest.php]: when I set it to true:
```console
Failed asserting that two strings are equal.
Expected :'<a href="https://mastodon.social/@morevnaproject" target="_blank">https://mastodon.social/@morevnaproject</a>'
Actual   :'https://mastodon.social/@morevnaproject'
```

**Comment by nupplaphil** [comment]
Yep, I mocked every dependency so far.
=> You can have a look at the `setup` function for the initial values I mock
=> L10n is returning the input as output

**Comment by MrPetovan** [comment]
Why did you leave some `htmlink` properties to `false` even though they should all pass as HTML links?

**Comment by nupplaphil** [comment]
> Why did you leave some `htmlink` properties to `false` even though they should all pass as HTML links?

Maybe the name of the attribute wasn't clear enough. The flag switches between "full" HTML output (`<a href...`) and "just" the link output.

For the two with flag "false", the output after the convert method is the same as the input.

**Comment by MrPetovan** [comment]
Ok, that's what I thought. Both those links need to be transformed to HTML links, they aren't counter-examples.

**Comment by nupplaphil** [comment]
Oh, OK now I understand your questions better

**Comment by MrPetovan** [comment]
I'm getting this weird error when running these tests: `PHP Fatal error:  Uncaught Exception: Serialization of 'Closure' is not allowed`

This error shows up with `bug-2487-3`, `bug-2487-4` and `bug-6095`, all the ones you disabled/changed.

**Comment by nupplaphil** [comment]
> I'm getting this weird error when running these tests: `PHP Fatal error: Uncaught Exception: Serialization of 'Closure' is not allowed`
> 
> This error shows up with `bug-2487-3`, `bug-2487-4` and `bug-6095`, all the ones you disabled/changed.

This is because of the annotation, remove the
```php
/**
  * @runTestsInSeparateProcesses
  * @preserveGlobalState disabled
  */
```

**Comment by MrPetovan** [comment]
Ok thanks, I have the expected failures now.

**Comment by MrPetovan** [comment]
Replaced by #6841

**Pull request closed without merging**